turbopack/crates/turbopack-node/src/worker_pool/operation.rs

```diff
@@ -0,0 +1,285 @@
+use std::{
+    process::ExitStatus,
+    sync::{Arc, LazyLock},
+};
+
+use anyhow::{Context, Result};
+use parking_lot::Mutex;
+use rustc_hash::FxHashMap;
+use tokio::sync::{
+    Mutex as AsyncMutex,
+    mpsc::{self, UnboundedReceiver, UnboundedSender},
+    oneshot,
+};
+use turbo_rcstr::RcStr;
+
+use crate::{evaluate::Operation, pool_stats::NodeJsPoolStats, worker_pool::worker_thread};
+
+#[derive(Clone)]
+pub(crate) struct MessageChannel<T: Send + Sync + 'static> {
+    sender: UnboundedSender<T>,
+    receiver: Arc<AsyncMutex<UnboundedReceiver<T>>>,
+}
+
+impl<T: Send + Sync + 'static> MessageChannel<T> {
+    pub(super) fn unbounded() -> Self {
+        let (sender, receiver) = mpsc::unbounded_channel();
+        Self {
+            sender,
+            receiver: Arc::new(AsyncMutex::new(receiver)),
+        }
+    }
+}
+
+impl<T: Send + Sync + 'static> MessageChannel<T> {
+    pub(crate) async fn send(&self, data: T) -> Result<()> {
+        self.sender
+            .send(data)
+            .map_err(|_| anyhow::anyhow!("failed to send message"))
+    }
+
+    pub(crate) async fn recv(&self) -> Result<T> {
+        let mut rx = self.receiver.lock().await;
+        rx.recv()
+            .await
+            .ok_or_else(|| anyhow::anyhow!("failed to recv message"))
+    }
+}
+
+pub(crate) struct PoolState {
+    pub(crate) idle_workers: Mutex<Vec<u32>>,
+    pub(crate) stats: Arc<Mutex<NodeJsPoolStats>>,
+    pub(crate) waiters: Mutex<Vec<oneshot::Sender<u32>>>,
+}
+
+impl Default for PoolState {
+    fn default() -> Self {
+        Self {
+            idle_workers: Mutex::new(Vec::new()),
+            stats: Arc::new(Mutex::new(NodeJsPoolStats::default())),
+            waiters: Mutex::new(Vec::new()),
+        }
+    }
+}
+
+#[derive(Default)]
+pub(super) struct PoolOptions {
+    pub(super) filename: RcStr,
+    pub(super) cwd: RcStr,
+}
+
+pub(crate) struct WorkerPoolOperation {
+    worker_routed_channel: Mutex<FxHashMap<u32, MessageChannel<(u32, String)>>>,
+    task_routed_channel: Mutex<FxHashMap<u32, MessageChannel<String>>>,
```

Comment:
```suggestion
    worker_routed_channel: Mutex<FxHashMap<u32, Arc<MessageChannel<(u32, String)>>>>,
    task_routed_channel: Mutex<FxHashMap<u32, Arc<MessageChannel<String>>>>,
```

This could avoid the clone of the MessageChannel and the inner Arcs.